src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -42,35 +46,136 @@ public void editPerson() {
+        EditPersonDialogHandle editPersonDialog =  personListPanel.clickEdit();
+        editPersonDialog.enterFirstName("Peter");
+        editPersonDialog.enterLastName("");
+        editPersonDialog.clickOk();
+
+        assertTrue(editPersonDialog.isInputValidationErrorDialogShown());
+        editPersonDialog.dissmissErrorMessage("Invalid Fields");
+        Assert.assertFalse(editPersonDialog.isInputValidationErrorDialogShown());
+        
+    }
+
+    @Test
+    public void cancelPerson_usingAccelerator() {
+
+        //Delete
+        personListPanel.clickOnPerson(td.alice);
+        PersonCardHandle deletedCard = personListPanel.getPersonCardHandle(new Person(personListPanel.getSelectedPerson()));
+        personListPanel.use_PERSON_DELETE_ACCELERATOR();
+        mainGui.sleep(1, TimeUnit.SECONDS);
+        assertTrue(deletedCard.isShowingGracePeriod("Deleting"));
+
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
+        mainGui.sleep(1, TimeUnit.SECONDS);
+        assertFalse(deletedCard.isShowingGracePeriod("Deleting"));
+        assertEquals(statusBar.getText(), "Delete Person [ " + deletedCard.getFirstName() + " "
+                     + deletedCard.getLastName() + " ] was cancelled.");
+
+
+        //Edit
+        Person newAlice = new PersonBuilder(td.alice.copy()).withFirstName("Alicia").withLastName("Brownstone")
+                .withStreet("Updated street").withCity("Singapore").withPostalCode("123123")
+                .withBirthday("01.01.1979").withGithubUsername("alicebrown123").withTags(td.colleagues, td.friends).build();
+
+        //Get a reference to the card displaying Alice's details
+        PersonCardHandle alicePersonCard = personListPanel.getPersonCardHandle(td.alice);
+
+        //Edit Alice to change to new values
+        personListPanel.clickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        editPersonDialog.enterNewValues(newAlice);
+        editPersonDialog.pressEnter();
+
+        mainGui.focusOnMainApp();
+        personListPanel.clickOnPerson(newAlice);
+
+        assertNotEquals(alicePersonCard, td.alice);
+        assertEquals(alicePersonCard, newAlice);
+        mainGui.sleep(ModelManager.GRACE_PERIOD_DURATION/2, TimeUnit.SECONDS);
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
+        assertEquals(alicePersonCard, td.alice);
+        assertEquals(statusBar.getText(), "Edit Person [ Alice Brown -> Alicia Brownstone ] was cancelled.");
+
+        //New
+        EditPersonDialogHandle addPersonDialog = personListPanel.clickNew();
+        Person pandaWong = new PersonBuilder(td.alice.copy()).withFirstName("Panda").withLastName("Wong")
+                .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
+                .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
```

Comment:
This builder method is used for creating a person similar to an existing person. We can write a separate builder method for creating an entirely new person.